toonz/sources/toonz/filmstrip.cpp

```diff
@@ -400,13 +402,31 @@ void FilmstripFrames::hideEvent(QHideEvent *) {
   disconnect(app->getCurrentOnionSkin(), SIGNAL(onionSkinMaskChanged()), this,
              SLOT(update()));
 
-  ComboViewerPanel *inknPaintViewerPanel = app->getInknPaintViewerPanel();
-  if (inknPaintViewerPanel) {
-    SceneViewer *viewer = inknPaintViewerPanel->getSceneViewer();
-    if (viewer) {
-      disconnect(viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
-      disconnect(viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
-    }
+  if (m_viewer) {
+    disconnect(m_viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
+    disconnect(m_viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
+  }
+}
+
+//-----------------------------------------------------------------------------
+
+void FilmstripFrames::getViewer() {
+  bool viewerChanged = false;
+  if (m_viewer && m_viewer != TApp::instance()->getActiveViewer()) {
+    disconnect(m_viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
+    disconnect(m_viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
+    viewerChanged = true;
+  }
+  ComboViewerPanel *inknPaintViewerPanel =
+      TApp::instance()->getInknPaintViewerPanel();
+  if (m_isGhibli && inknPaintViewerPanel) {
+    m_viewer = inknPaintViewerPanel->getSceneViewer();
+  } else {
+    m_viewer = TApp::instance()->getActiveViewer();
+  }
+  if (m_viewer) {
+    connect(m_viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
+    connect(m_viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
```

Comment:
For now, this function `getViewer()` connects signals every time you enter the cursor in the Level Strip and causes multiple connection.
(Just as a side note, you can dump the connection info with `QObject::dumpObjectInfo()` under the debug mode)
I think you meant (and forgot) to use `viewerChanged` flag for avoiding such behavior.

How about modifying this function like as follows? Please note that the former block of modification is for connecting the signals properly on launch:

```
@@ -412,9 +412,11 @@ void FilmstripFrames::hideEvent(QHideEvent *) {
 
 void FilmstripFrames::getViewer() {
   bool viewerChanged = false;
-  if (m_viewer && m_viewer != TApp::instance()->getActiveViewer()) {
-    disconnect(m_viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
-    disconnect(m_viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
+  if (m_viewer != TApp::instance()->getActiveViewer()) {
+    if (m_viewer) {
+      disconnect(m_viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
+      disconnect(m_viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
+    }
     viewerChanged = true;
   }
   ComboViewerPanel *inknPaintViewerPanel =
@@ -424,7 +426,7 @@ void FilmstripFrames::getViewer() {
   } else {
     m_viewer = TApp::instance()->getActiveViewer();
   }
-  if (m_viewer) {
+  if (m_viewer && viewerChanged) {
     connect(m_viewer, SIGNAL(onZoomChanged()), this, SLOT(update()));
     connect(m_viewer, SIGNAL(refreshNavi()), this, SLOT(update()));
   }
```